src/main/kotlin/no/nav/tilleggsstonader/sak/hendelser/personhendelse/dødsfall/DødsfallHåndterer.kt

```diff
@@ -0,0 +1,137 @@
+    private val vedtaksperiodeService: VedtaksperiodeService,
+    private val hendelseRepository: HendelseRepository,
+    private val oppgaveService: OppgaveService,
+) {
+    companion object {
+        private const val ANNULLERT_DØDSFALL_BESKRIVELSE = "\n\nDenne oppgaven har blitt annullert"
+    }
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Transactional
+    fun håndter(dødsfallHendelse: DødsfallHendelse) {
+        if (dødsfallHendelse.erAnnullering) {
+            håndterAnnullertDødsfall(dødsfallHendelse.hendelseId)
+        } else if (hendelseRepository.existsByTypeAndId(TypeHendelse.PERSONHENDELSE, dødsfallHendelse.hendelseId)) {
+            logger.info("Dødsfallhendelse med id ${dødsfallHendelse.hendelseId} er allerede behandlet")
+        } else {
+            håndterDødsfall(dødsfallHendelse)
+        }
+    }
+
+    private fun håndterDødsfall(dødsfallHendelse: DødsfallHendelse) {
+        val fagsakerMedAktiveVedtak = finnFagsakerMedLøpendeUtbetalingFraDødsdato(dødsfallHendelse)
+
+        if (fagsakerMedAktiveVedtak.isNotEmpty()) {
+            val opprettOppgaveTask = fagsakerMedAktiveVedtak.opprettTask(dødsfallHendelse)
+
+            logger.info("Oppretter task for opprettelse av oppgave for håndtering av dødsfall som kjører ${opprettOppgaveTask.triggerTid}")
+            val lagretTask = taskService.save(opprettOppgaveTask)
+
+            hendelseRepository.insert(
+                Hendelse(
+                    type = TypeHendelse.PERSONHENDELSE,
+                    id = dødsfallHendelse.hendelseId,
+                    metadata = mapOf("taskId" to lagretTask.id.toString()),
+                ),
+            )
+        }
+    }
+
+    private fun finnFagsakerMedLøpendeUtbetalingFraDødsdato(dødsfallHendelse: DødsfallHendelse) =
+        fagsakService
+            .finnFagsaker(dødsfallHendelse.personidenter.toSet())
+            .filter { harAktivtVedtak(it, dødsfallHendelse.dødsdato) }
+
+    private fun List<Fagsak>.opprettTask(dødsfallHendelse: DødsfallHendelse): Task {
+        val fagsak = first()
+        logger.info("Oppretter oppgave for håndtering av dødsfall for person ${fagsak.id}")
+
+        // TODO - ved nye stønadstyper som ikke behandles av NAY bør vi lagre enhet på fagsak og opprette oppgave til alle enheter
+        return OpprettDødsfallOppgaveTask
+            .opprettTask(
+                stønadstype = fagsak.stønadstype,
+                dødsfallHendelse = dødsfallHendelse,
+            ).copy(triggerTid = LocalDateTime.now().plusWeeks(1))
+    }
+
+    private fun harAktivtVedtak(
+        fagsak: Fagsak,
+        dødsdato: LocalDate,
+    ): Boolean {
+        val behandling = behandlingService.finnSisteIverksatteBehandling(fagsak.id)
+        if (behandling != null) {
+            return vedtaksperiodeService
+                .finnVedtaksperioderForBehandling(behandling.id, null)
+                .any { it.inneholder(dødsdato) }
```

Comment:
Burde man sjekke om dødsdato er før en periode? 
Eks hvis personen har en periode frem i tiden, men ikke akkurat for datoet man dør
```suggestion
                .any { it.tom > dødsdato }
```

Reply:
:+1: Ja, må fikse opp i noe logikk her..